text/0000-new-test-waiters.md

```diff
@@ -58,7 +58,7 @@ registerWaiter(function() {
 
 While reading the above is straightforward, when writing a test waiter using the old system it's easy to forget what the expected return value is: `true` or `false`. Additionally, it's a bit more cognitive overhead to derive what the intended result of the particular boolean return value is: does returning `true` result in the test waiter waiting or not?
 
-As mentioned before, there's no additional information provided via `registerWaiter`, and no way to capture stack traces at the call site. Unmanaged async that 'hangs' can cause your tests to stall and ultimately timeout. This is particularly problematic when trying to identify which of many test waiters has caused this timeout.
+As mentioned before, there's no additional information provided via `registerWaiter`, and capturing stack traces at the call site is currently not implemented. Unmanaged async that 'hangs' can cause your tests to stall and ultimately timeout. Not having stack traces is particularly problematic when trying to identify which of many test waiters has caused this timeout, as it's like looking for a needle in a haystack. The new system captures the stacks lazily, when the waiter's `beginAsync` method is called (more on `beginAsync` later). This allows for identifying the offending code more easily.
```

Comment:
> The new system captures the stacks lazily, when the waiter's `beginAsync` method is called (more on `beginAsync` later).

I think this phrasing is _slightly_ confusing. We eagerly create the error object, but only force the stack to be reified when it's asked for (e.g. via `getSettledState`). I'm not sure how to say this better than what you have here though. 🤔

Reply:
I also felt the same. I have an idea. Will update.

Reply:
Updated. Take a look and let me know what you think.